storage/tianmu/core/tianmu_attr.cpp

```diff
@@ -87,19 +87,42 @@ void TianmuAttr::Create(const fs::path &dir, const AttributeTypeInfo &ati, uint8
   fmeta.WriteExact(&meta, sizeof(meta));
   fmeta.Flush();
 
+  size_t no_nulls{no_rows};
+  int64_t min{0};
+  int64_t max{0};
+  bool int_not_null{false};
```

Comment:
why not use AttributeTypeInfo::NotNull() to check whether the field is NULL? I guess int_not_null means has_default_value?

Reply:
1.I do not to whether not null flag is proper, in the scenario the alter table defined not null or null(same with tianmu), the
output is NULL. I just want to use GetValueFromField(it deals the default value logic) function to deal with the scenario, the std::variants store the value,
2 it has the interface to decide whether it is null,if not null, what is the default value.during the debug, it seems the not null flag is not set such as the scenario as below( i am not sure).
3 I think the current solution is also a proper way,not to consider the null flag (or it has been considered in GetValueFromField)
4 the std::variant in gcc8 has bugs, what we supported is gcc9.3, if we do not upgrade debian gcc to 9.3, another solution is to use boost:variant, but not recommended. It is better to upgrade the debian CI to gcc9.3 @wisehead 

```
create table t4(id int,name varchar(5)) ;
insert into t4 values(1,'AAA'),(2,'BBB');
alter table t4 add column age DECIMAL(17,9);
select * from t4;
id	name	age
1	AAA	NULL
2	BBB	NULL
```